### ISSUE TITLE
TypeScript 5.4 aggressive union type narrowing breaks svelte-check

### ISSUE THREAD

**Issue by AaronAtP123** [sveltejs/language-tools#2316]
### Describe the bug

TypeScript's new type narrowing behavior is causing svelte-check to emit `This comparison appears to be unintentional because the types '...' and '...' have no overlap.` on a variable when its value is mutated via binding.

Note this change is only currently affecting top-level variables. Union types also include boolean and enum types.

### Reproduction

```svelte
<script lang="ts">let checked = true;</script>
<input type="checkbox" bind:checked />
{#if checked === false}false{/if}
```

### Expected behaviour

svelte-check should pass with no errors. Since bound expressions are type checked as covariant to the prop's type, the target variable's effective type ought to be widened to the entire union type.

### System Info

svelte-check 3.6.7
- OS: Windows 10
- IDE: VSCode

### Which package is the issue about?

svelte-check

**Comment by enyo** [comment]
Just ran into this. This fails for me:

```svelte
<script lang="ts">
  let period: 'monthly' | 'yearly' = 'yearly'
</script>

{#if period === 'monthly'}
  paid every month
{/if}
```

With this error:

```
Error: This comparison appears to be unintentional because the types '"yearly"' and '"monthly"' have no overlap. (ts)
```

**Comment by 4750moorem** [comment]
> Just ran into this. This fails for me:
> 
> ```svelte
> <script lang="ts">
>   let period: 'monthly' | 'yearly' = 'yearly'
> </script>
> 
> {#if period === 'monthly'}
>   paid every month
> {/if}
> ```
> 
> With this error:
> 
> ```
> Error: This comparison appears to be unintentional because the types '"yearly"' and '"monthly"' have no overlap. (ts)
> ```
> 
> **EDIT**: To temporarily work around this issue, I changed the typescript dependency to: `5.3.x`

You can cast the string 'yearly' as 'monthly' | 'yearly' to get around it for now if you wanna stay on 5.4.x.

```
 let period: 'monthly' | 'yearly' = 'yearly' as 'monthly' | 'yearly' 
 ```

**Comment by jasonlyu123** [comment]
This one is expected because there isn't any reassignment. If your actual case has an assignment somewhere that isn't binding. Can you share it here?

**Comment by danieldiekmeier** [comment]
@dummdidumm Sadly, this is not completely fixed in the latest svelte-check v3.6.8. Maybe it's missing a special case for `bind:group`? At least that's the problem I'm running into. I built a small reproduction: https://stackblitz.com/edit/vitejs-vite-r64k1v?file=src%2FApp.svelte

If you run `pnpm i && pnpm check`, you can still see the error.
